src/main.c

```diff
@@ -206,6 +219,9 @@ int main(int argc, char *argv[]) {
 			case 't':
 				ctx.clientmgr_ctx.export_table = atoi(optarg);
 				break;
+			case 's':
+				socket_init(&ctx.socket_ctx, optarg);
```

Comment:
I don't like the way this initializes two sockets (leaking the first one) when -s is given multiple times. We should just save a pointer to the path in a variable and call *socket_init* with that after the getopt loop. I think setting fd the -1 should rather be done in *socket_init* when the given path is NULL.